### DISCUSSION TITLE
JSR does not support `viem`'s strong types

### DISCUSSION THREAD

**Discussion by allezxandre** [wevm/viem#3060] [Idea / Feature Request] [answered]
### Check existing issues

- [X] I checked there isn't [already an issue](https://github.com/wevm/viem/issues) for the bug I encountered.

### Viem Version

2.21.50

### Current Behavior

The JSR page of `viem` does not provide types for some modules (see [the documentation for the `mainnet` chain definition](https://jsr.io/@wevm/viem@2.21.50/doc/chains/~/mainnet)).

In editors like VSCode, this means that most types imported from JSR are cast to `any`, and do not leverage the type safety of Viem.

### Anything else?

It is my understanding that it comes from what JSR calls ["slow types"](https://jsr.io/docs/about-slow-types), and there isn't much we can do about it for now. 

Fortunately, we can still use the package published on `npm`. But I'm curious to know if there are any plans going forward to make `viem` work with JSR, or if JSR's no-slow types policy is just too much at odds with Viem's strong type guarantees.

**Comment by allezxandre** [comment]
I should also add for the sake of completeness that TypeScript 5.5 has separately introduced an [`--isolatedDeclarations`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-5.html#isolated-declarations) compiler option that goes in the direction of JSR's slow types policy.

**Comment by tmm** [comment, marked as answer]
Not much we can do about this at the moment. The "Slow types" idea seems great, but libraries that do more in-depth inference (like Viem, [tRPC](https://jsr.io/@trpc/server/score), and others) won't be able to adhere in it's current state without compromising developer experience.
